packages/amplify_core/lib/src/types/notifications/push/push_notification_message.dart

```diff
@@ -30,42 +30,50 @@ class PushNotificationMessage
-        title = alert['title'] as String?;
-        body = alert['body'] as String?;
-        imageUrl = data['media-url'] as String?;
+      final aps = json['aps'] as Map<Object?, Object?>?;
+      if (aps != null) {
+        final alert = aps['alert'] as Map<Object?, Object?>?;
+        if (alert != null) {
+          title = alert['title'] as String?;
+          body = alert['body'] as String?;
+          imageUrl = data['media-url'] as String?;
 
-        if (data['pinpoint'] != null) {
-          final pinpointData = data['pinpoint'] as Map<Object?, Object?>;
-          deeplinkUrl = pinpointData['deeplink'] as String?;
-          goToUrl = deeplinkUrl;
+          if (data['pinpoint'] != null) {
+            final pinpointData = data['pinpoint'] as Map<Object?, Object?>;
+            deeplinkUrl = pinpointData['deeplink'] as String?;
+            goToUrl = deeplinkUrl;
+          }
+          apnsOptions =
+              ApnsPlatformOptions(subtitle: aps['subtitle'] as String?);
+        }
+      } else {
+        final action = json['action'] as Map<Object?, Object?>?;
+        title = json['title'] as String?;
+        body = json['body'] as String?;
+        imageUrl = json['imageUrl'] as String?;
+        if (action != null) {
+          deeplinkUrl = action['deeplink'] as String?;
+          goToUrl = action['url'] as String?;
+        }
+        final fcmOptionsMap = json['fcmOptions'] as Map<Object?, Object?>?;
+        if (fcmOptionsMap != null) {
+          final sentTimeInt = fcmOptionsMap['sentTime'] as int?;
+          final sentTime = sentTimeInt == null
+              ? null
+              : DateTime.fromMillisecondsSinceEpoch(sentTimeInt);
+          fcmOptions = FcmPlatformOptions(
+            channelId: fcmOptionsMap['channelId'] as String?,
+            messageId: fcmOptionsMap['messageId'] as String?,
+            senderId: fcmOptionsMap['sender'] as String?,
+            sentTime: sentTime,
+          );
         }
-        apnsOptions = ApnsPlatformOptions(subtitle: aps['subtitle'] as String?);
-      }
-    } else {
-      final fcmOptionsMap = json['fcmOptions'] as Map<Object?, Object?>?;
-      title = json['title'] as String?;
-      body = json['body'] as String?;
-      imageUrl = json['imageUrl'] as String?;
-      deeplinkUrl = json['deeplinkUrl'] as String?;
-      goToUrl = json['goToUrl'] as String?;
-      if (fcmOptionsMap != null) {
-        final sentTimeInt = fcmOptionsMap['sentTime'] as int?;
-        final sentTime = sentTimeInt == null
-            ? null
-            : DateTime.fromMillisecondsSinceEpoch(sentTimeInt);
-        fcmOptions = FcmPlatformOptions(
-          channelId: fcmOptionsMap['channelId'] as String?,
-          messageId: fcmOptionsMap['messageId'] as String?,
-          senderId: fcmOptionsMap['sender'] as String?,
-          sentTime: sentTime,
-        );
       }
+    } on Error catch (e) {
+      safePrint('Error when converting Json to PushNotificationMessage: $e');
```

Comment:
What's the source of error here if the code has already handled fields nullability correctly?

Reply:
I added it to gracefully fail if the type of any field changes. Since we pass the rawData and other fields as we get in the native code, the type of them may be changed by cx when composing their messages I think. So, catching it and printing it rather than crashing. 

What would be the alternative here?